src/mptool.f90

```diff
@@ -132,7 +152,134 @@ real(mytype) function  pmax_mytype(var)
                                                     mpi_comm_world,ierr)
     !
   end function pmax_mytype
+
+  real(mytype) function  pmax_mytype_array(var)
+    !
+    ! arguments
+    real(mytype),intent(in) :: var(:)
+    !
+    ! local data
+    integer :: ierr
+    !
+    call mpi_allreduce(var,pmax_mytype_array,size(var),real_type,mpi_max,    &
+                                                    mpi_comm_world,ierr)
+    !
+  end function pmax_mytype_array
+
+  real(mytype) function  pmin_mytype_array(var)
+    !
+    ! arguments
+    real(mytype),intent(in) :: var(:)
+    !
+    ! local data
+    integer :: ierr
+    !
+    call mpi_allreduce(var,pmin_mytype_array,size(var),real_type,mpi_min,    &
+                                                    mpi_comm_world,ierr)
+    !
+  end function pmin_mytype_array
+  real(mytype) function  pmin_mytype(var)
+    !
+    ! arguments
+    real(mytype),intent(in) :: var
+    !
+    ! local data
+    integer :: ierr
+    !
+    call mpi_allreduce(var,pmin_mytype,1,real_type,mpi_min,    &
+                                                    mpi_comm_world,ierr)
+    !
+  end function pmin_mytype
+  !
+  !+-------------------------------------------------------------------+
+  !| this function is to update table based on alltoall mpi            |
+  !+-------------------------------------------------------------------+
+  !| CHANGE RECORD                                                     |
+  !| -------------                                                     |
+  !| 17-Jun-2022  | Created by J. Fang STFC Daresbury Laboratory       |
+  !+-------------------------------------------------------------------+
+  function ptable_update_int_arr(vain) result(vout)
+    !
+    use mpi
+    !
+    integer,intent(in) :: vain(:)
+    integer :: vout(size(vain))
+    !
+    ! local variables
+    integer :: nvar,ierr
+    !
+    nvar=size(vain)
+    !
+    call mpi_alltoall(vain,1,mpi_integer,                   &
+                      vout,1,mpi_integer,mpi_comm_world,ierr)
+    !
+    return
```

Comment:
Return statement is not needed at the end of a function / subroutine

Reply:
I sorted this